src/components/AcceptedValuesField/AcceptedValuesField.js

```diff
@@ -53,18 +54,32 @@ export const AcceptedValuesField = ({
     name: optionTemplate ? updateValueWithTemplate(option, optionTemplate) : option.name,
   }));
 
+  const extendDataWithStatisticalCodeType = (arrToExtend, arrWithExtendedField) => arrToExtend.map(item => {
+    const correspondRecord = arrWithExtendedField.find(itemAlt => itemAlt.id === item.statisticalCodeTypeId);
+
+    return {
+      ...item,
+      statisticalCodeTypeName: correspondRecord.name,
+    };
+  });
+
+  const mappedFns = { statCodes: extendDataWithStatisticalCodeType };
+
   useEffect(() => {
-    if (wrapperSourceLink && wrapperSourcePath && isEmpty(acceptedValuesList)) {
-      fetchAcceptedValuesList(okapi, wrapperSourceLink, wrapperSourcePath)
-        .then(data => {
-          const acceptedValues = getAcceptedValuesObj(data);
-          const updatedListOptions = updateListOptions(data);
-
-          setListOptions(updatedListOptions);
-          setAcceptedValues(acceptedValues);
-        });
+    if (wrapperSources && isEmpty(acceptedValuesList)) {
+      const promises = wrapperSources.map(source => fetchAcceptedValuesList(okapi, source.wrapperSourceLink, source.wrapperSourcePath));
+
+      Promise.all(promises).then(result => {
+        const dataWithExtendField = wrapperSourcesFn ? sortCollection(mappedFns[wrapperSourcesFn](sortCollection(result[0], ['code']), result[1]), ['statisticalCodeTypeName']) : '';
```

Comment:
Since this code is applied for every wrappedSource, in this case, I assume `statisticalCodeTypeName` is not relevant for all cases, right?

Reply:
Great comment, thank you! Fixed